src/Aspire.Hosting.Azure.ServiceBus/AzureServiceBusExtensions.cs

```diff
@@ -275,7 +278,7 @@ public static IResourceBuilder<AzureServiceBusResource> RunAsEmulator(this IReso
         ServiceBusClient? serviceBusClient = null;
         string? queueOrTopicName = null;
 
-        builder.ApplicationBuilder.Eventing.Subscribe<AfterEndpointsAllocatedEvent>(async (@event, ct) =>
+        builder.ApplicationBuilder.Eventing.Subscribe<BeforeResourceStartedEvent>(async (@event, ct) =>
```

Comment:
You only want the code to run for the current resource, so you need to do something like this:

```suggestion
        builder.ApplicationBuilder.Eventing.Subscribe<BeforeResourceStartedEvent>(builder.Resource, async (@event, ct) =>
```